representation/transparency_using_selection/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import { createRootViewer } from "./common/init";
+import { setStructureTransparency } from "molstar/lib/mol-plugin-state/helpers/structure-transparency";
+import { Queries, QueryContext, StructureProperties, StructureSelection } from "molstar/lib/mol-model/structure";
+
+async function init() {
+    // Create viewer
+    const plugin = await createRootViewer();
+    
+    // Download PDB
+    const fileData = await plugin.builders.data.download(
+        { url: "https://models.rcsb.org/4hhb.bcif", isBinary: true }
+    );
+
+    // Load PDB and create representation
+    const trajectory = await plugin.builders.structure.parseTrajectory(fileData, "mmcif");
+    const data = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
+
+    // Get Structure and StructureRef objects from data
+    const struct = data?.structure.cell?.obj?.data!;
+    const structRef = plugin.managers.structure.hierarchy.current.structures.find((sr)=>sr.cell.obj?.data === struct)
+    
+    // Query all chains with ID A or B
+    const ctx = new QueryContext(struct)
+    const query = Queries.generators.chains({
+        chainTest: l => StructureProperties.chain.label_asym_id(l.element) == 'A' || StructureProperties.chain.label_asym_id(l.element)== 'B'
+    })
+    const res = query(ctx);
+    // Convert the StructureSelection into a Loci
+    // Set a transparency of 1 (invisible) on the components using the loci
+    setStructureTransparency(plugin, structRef?.components!, 1, async ()=> StructureSelection.toLociWithSourceUnits(res))
```

Comment:
There is something I don't get in that example: we hide everything that's from chain A or B. How come it's only the protein chains that are hidden? (the hemes and water molecules remain)

Reply:
maybe its related to the SourceUnit thing?

Reply:
For the source unit thing, it is almost always the case that we want to use `toLociWithSourceUnits` instead of `toLociWithCurrentUnits`. The mol* code almost always uses the `toLociWithSourceUnits` (except in 1 example where it is doing a superposition test)
- Source units uses the `StructureSelection.source`'s units for the loci elements.
- Current units uses the `.source` of a union of all the structures in the `StructureSelection`.
It is a little confusing and the effective difference isn't clear, but I think it has something to do with manipulating the source units directly or a copy of them (you would want to use a copy in a test).

For the reason only the protein chains are hidden, in this specific PDB (4hhb), there are 14 chains (A through N) and  5 entities. Chains A and B are protein chains, there are multiple water chains and heme chains. It is specifically how this PDB is written

Reply:
For the source unit thing, there isn't any `.toLoci` function for a `StructureSelection`. We need to use either SourceUnits or CurrentUnits. It is almost always the case that we want to use `toLociWithSourceUnits` instead of `toLociWithCurrentUnits`. The mol* code almost always uses the `toLociWithSourceUnits` (except in 1 example where it is doing a superposition test)
- Source units uses the `StructureSelection.source`'s units for the loci elements.
- Current units uses the `.source` of a union of all the structures in the `StructureSelection`.
It is a little confusing and the effective difference isn't clear, but I think it has something to do with manipulating the source units directly or a copy of them (you would want to use a copy in a test).

For the reason only the protein chains are hidden, in this specific PDB (4hhb), there are 14 chains (A through N) and  5 entities. Chains A and B are protein chains, there are multiple water chains and heme chains. It is specifically how this PDB is written

Reply:
Oh, I see. That's the difference between the `label_asym_id` and the `auth_asym_id` which has only chains defined from A to D.
<img width="1051" alt="image" src="https://github.com/molstar/example-gallery/assets/116235895/984c1597-f871-406d-bbd2-ea6bc590147f">